src/mono/CMakeLists.txt

```
@@ -113,14 +113,7 @@ endif()
 set(HAVE_MOVING_COLLECTOR 1)
 set(HAVE_CONC_GC_AS_DEFAULT 1)
 set(MONO_INSIDE_RUNTIME 1)
-
-TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
-
-if(IS_BIG_ENDIAN)
-  set(TARGET_BYTE_ORDER G_BIG_ENDIAN)
-else()
-  set(TARGET_BYTE_ORDER G_LITTLE_ENDIAN)
-endif()
+set(TARGET_BYTE_ORDER CMAKE_CXX_BYTE_ORDER)
```

```suggestion
set(TARGET_BYTE_ORDER CMAKE_C_BYTE_ORDER)
```

Mono doesn't always enable C++, so use the `C` variant.

---

this isn't enough since we need to set `TARGET_BYTE_ORDER` to either `G_BIG_ENDIAN` or `G_LITTLE_ENDIAN`.

I'd suggest changing the check to:
```
if(CMAKE_C_BYTE_ORDER STREQUAL "BIG_ENDIAN")
...
```